src/blocks/posts/edit.js

```diff
@@ -11,22 +11,67 @@ const {
 	Spinner
 } = wp.components;
 
-const { useSelect } = wp.data;
+const { useSelect, registerGenericStore, dispatch } = wp.data;
+
+const { Fragment, useEffect, useState } = wp.element;
 
-const { Fragment } = wp.element;
 
 /**
  * Internal dependencies
  */
 import { StyleSwitcherBlockControl } from '../../components/style-switcher-control/index.js';
 import Inspector from './inspector.js';
 import Layout from './components/layout/index.js';
+import { getCustomPostTypeSlugs } from '../../helpers/helper-functions.js';
+
+
+/**
+ * Store
+ *
+ */
+
+const createPostStore = () => {
+	let storeChanged = () => {};
+	const store = { slugs: [] };
+
+	const selectors = {
+		getSlugs() {
+			return store.slugs;
+		}
+	};
+
+	const actions = {
+		setSlugs( newSlugs ) {
+			store.slugs = newSlugs;
+			storeChanged();
+		}
+	};
+
+	return {
+		getSelectors() {
+			return selectors;
+		},
+		getActions() {
+			return actions;
+		},
+		subscribe( listener ) {
+			storeChanged = listener;
+		}
+	};
+};
+
+registerGenericStore( 'otter-posts-block', createPostStore() );
```

Comment:
Instead of having many stores in each block, we can have one plugin-wide store that we can use.